eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+from collections import defaultdict
+from functools import cached_property
+from itertools import cycle
+
+from ebi_eva_common_pyutils.assembly import NCBIAssembly
+from ebi_eva_common_pyutils.config_utils import get_mongo_uri_for_eva_profile
+from ebi_eva_common_pyutils.logger import logging_config, AppLogger
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.mongodb import MongoDatabase
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+from ebi_eva_common_pyutils.taxonomy.taxonomy import normalise_taxon_scientific_name, get_scientific_name_from_ensembl
+
+
+# round-robin through the instances from 1 to 10
+tempmongo_instances = cycle([f'tempmongo-{instance}' for instance in range(1, 11)])
+
+all_tasks = ['fill_release_entries', 'fill_should_be_released']
+
+
+class ReleaseTracker(AppLogger):
+
+    def __init__(self, private_config_xml_file, maven_profile, release_version, reference_directory):
+        self.private_config_xml_file = private_config_xml_file
+        self.maven_profile = maven_profile
+        self.release_version = release_version
+        self.ref_dir = reference_directory
+
+    @cached_property
+    def metadata_conn(self):
+        return get_metadata_connection_handle(self.maven_profile, self.private_config_xml_file)
+
+    @cached_property
+    def mongo_conn(self):
```

Comment:
Previously the mongo database being used was _always_ production, even if you were creating the table in `development` - this seemed messy/confusing to me so I've unified it to use a single profile, but it did have the advantage that you could create a table in dev based on real production data for `should_be_released`. I don't think it matters much for how we use the table now (e.g. we don't actually compute any counts that need to be checked), but let me know if you think this functionality should be restored.